applications/ContactStructuralMechanicsApplication/custom_utilities/interface_preprocess.cpp

```diff
@@ -139,6 +143,122 @@ void InterfacePreprocessCondition::GenerateInterfacePart<3>(
+                CopyProperties(p_prop, p_prop_new, THICKNESS, false);
+                CopyProperties(p_prop, p_prop_new, YOUNG_MODULUS);
+
+                counter++;
+                break;
+            }
+        }
+
+        // Now we iterate over the conditions
+        if (counter > 0) {
+            ConditionsArrayType& conditions_array = rInterfacePart.Conditions();
+
+            #pragma omp parallel for
+            for(int i = 0; i < static_cast<int>(conditions_array.size()); ++i) {
+                auto it_cond = conditions_array.begin() + i;
+                it_cond->SetProperties(p_prop_new);
+            }
+        } else {
+            KRATOS_ERROR << "It was not possible to add a property" << std::endl;
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+bool InterfacePreprocessCondition::CheckOnTheFace(
+    const std::vector<std::size_t>& rIndexVector,
+    GeometryType& rElementGeometry
+    )
+{
+    if (rElementGeometry.WorkingSpaceDimension() == 2) {
+        for (IndexType i_edge = 0; i_edge < rElementGeometry.EdgesNumber(); ++i_edge) {
+            const IndexType number_of_nodes = rElementGeometry.Edges()[i_edge].size();
+            std::vector<IndexType> index_vector_face(number_of_nodes);
+            if (number_of_nodes == rIndexVector.size()) {
+                for (IndexType i_node = 0; i_node < number_of_nodes; i_node++) {
+                    index_vector_face[i_node] = rElementGeometry.Edges()[i_edge][i_node].Id();
+                }
+                std::sort(index_vector_face.begin(), index_vector_face.end());
+
+                bool is_here = false;
+                for (IndexType i_node = 0; i_node < number_of_nodes; i_node++) {
+                    if (index_vector_face[i_node] == rIndexVector[i_node]) is_here = true;
+                }
+                if (is_here == true) return true;
+            }
+        }
+    } else {
+        if (rElementGeometry.LocalSpaceDimension() == 3) {
+            for (IndexType i_face = 0; i_face < rElementGeometry.FacesNumber(); ++i_face) {
+                const IndexType number_of_nodes = rElementGeometry.Faces()[i_face].size();
+                if (number_of_nodes == rIndexVector.size()) {
+                    std::vector<IndexType> index_vector_face(number_of_nodes);
+                    for (IndexType i_node = 0; i_node < number_of_nodes; i_node++) {
+                        index_vector_face[i_node] = rElementGeometry.Faces()[i_face][i_node].Id();
+                    }
+
+                    std::sort(index_vector_face.begin(), index_vector_face.end());
+
+                    bool is_here = false;
+                    for (IndexType i_node = 0; i_node < number_of_nodes; i_node++) {
+                        if (index_vector_face[i_node] == rIndexVector[i_node]) is_here = true;
```

Comment:
you could break after the first true

Reply:
Nope, because in this case we check the id one by one, we only say is false in case one of them is false. I can break with the false, but I have concerns if the break breaks everything or just the last level

Reply:
I can change that in a future PR